driver-sync/src/test/functional/com/mongodb/client/AbstractSessionsProseTest.java

```diff
@@ -103,4 +113,79 @@ public void commandStarted(final CommandStartedEvent event) {
+                                @Override
+                                public void commandStarted(final CommandStartedEvent event) {
+                                    containsLsid.set(event.getCommand().containsKey("lsid"));
+                                }
+                            })
+                            .build())) {
+
+                MongoCollection<Document> collection = client.getDatabase(getDefaultDatabaseName()).getCollection(getClass().getName());
+                try {
+                    collection.find().first();
+                } catch (MongoCommandException e) {
+                    // ignore command errors from mongocryptd
+                }
+                assertFalse(containsLsid.get());
+            }
+        } finally {
+            mongocryptdProcess.destroy();
+        }
+    }
+
+    @Test
+    public void shouldThrowOnExplicitSessionIfConnectionDoesNotSupportSessions() throws IOException {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        Process mongocryptdProcess = startMongocryptdProcess("2");
+        try {
+            try (MongoClient client = getMongoClient(getMongocryptdMongoClientSettingsBuilder().build())) {
+                MongoCollection<Document> collection = client.getDatabase(getDefaultDatabaseName()).getCollection(getClass().getName());
+                try (ClientSession session = client.startSession()) {
+                    String expectedClientExceptionMessage =
+                            "Attempting to use a ClientSession while connected to a server that doesn't support sessions";
+                    try {
+                        collection.find(session).first();
+                        fail("Expected MongoClientException");
+                    } catch (MongoClientException e) {
+                        assertEquals(expectedClientExceptionMessage, e.getMessage());
+                    }
+
+                    try {
+                        collection.insertOne(session, new Document());
+                        fail("Expected MongoClientException");
+                    } catch (MongoClientException e) {
+                        assertEquals(expectedClientExceptionMessage, e.getMessage());
+                    }
+                }
+            }
+        } finally {
+            mongocryptdProcess.destroy();
+        }
+    }
+
+    private static MongoClientSettings.Builder getMongocryptdMongoClientSettingsBuilder() {
+        return MongoClientSettings.builder()
+                .applyToClusterSettings(builder ->
+                        builder.hosts(singletonList(new ServerAddress("localhost", MONGOCRYPTD_PORT))));
+    }
+
+    private static Process startMongocryptdProcess(final String pidSuffix) throws IOException {
+        ProcessBuilder processBuilder = new ProcessBuilder(asList("mongocryptd",
+                "--port", Integer.toString(MONGOCRYPTD_PORT),
+                "--pidfilepath", "mongocryptd-" + pidSuffix + ".pid"));
```

Comment:
This was a bit hard to figure out.  Because other FLE tests have already started mongocryptd and it's likely still running, and mongocryptd will shut down if there is an existing pid file, we have to specify a different pidfilepath for this mongocrypt to prevent it from immediately shutting down.

Reply:
Ack